--- platform.rb.old     2011-10-04 10:23:07.000000000 +0100
+++ platform.rb 2011-10-05 16:33:03.000000000 +0100
@@ -28,8 +28,14 @@
 require_plugin 'linux::lsb'

 if lsb[:id]
-  platform lsb[:id].downcase
-  platform_version lsb[:release]
+  if lsb[:id] =~ /RedHat/i
+    contents = File.read("/etc/redhat-release").chomp
+    platform get_redhatish_platform(contents)
+    platform_version get_redhatish_version(contents)
+  else
+    platform lsb[:id].downcase
+    platform_version lsb[:release]
+  end
 elsif File.exists?("/etc/debian_version")
   platform "debian"
   platform_version File.read("/etc/debian_version").chomp
